File: chess/Rook.py
from .ChessPiece import ChessPiece

class Rook(ChessPiece):
    def __init__(self, player, pos):
        super().__init__(player, pos)
        self.multipleMove = True
        self.canJumpOverOthers = False
        self.possibleRegularMoves = self.possibleCaptureMoves = [(0, 1), (1, 0), (0, -1), (-1, 0)]
        self.name = 'Rook'

File: chess++/Rook.cpp
#include "Rook.h"

Rook::Rook(const Player& player, const Position& position)
: ChessPiece(player, position) {
    this->name = "Rook";
    this->multipleMove = true;
    this->canJumpOverOthers = false;

    // Initializing possible moves for a rook
    this->possibleRegularMoves = {
        {0, 1}, {1, 0}, {0, -1}, {-1, 0}
    };
    // For Rook, capture moves are the same as regular moves
    this->possibleCaptureMoves = this->possibleRegularMoves;
}